File: backend/src/main/java/io/blog/devlog/domain/mail/service/CommentEmailSubService.java
package io.blog.devlog.domain.mail.service;

import com.fasterxml.jackson.databind.ObjectMapper;
import io.blog.devlog.domain.mail.types.MessageType;
import io.blog.devlog.global.redis.message.CommentEmailMessage;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.data.redis.connection.Message;
import org.springframework.data.redis.connection.MessageListener;
import org.springframework.lang.Nullable;
import org.springframework.stereotype.Service;

import java.util.HashMap;

@Slf4j
@Service
@RequiredArgsConstructor
public class CommentEmailSubService implements MessageListener {

    private final EmailService emailService;
    private final ObjectMapper mapper;

    @Override
    public void onMessage(Message message, @Nullable byte[] pattern) {
        try {
            CommentEmailMessage emailMessage = mapper.readValue(message.getBody(), CommentEmailMessage.class);

            log.info("Received message from Redis");
            log.info("Email: {}", emailMessage.getEmail());
            log.info("Subject: {}", emailMessage.getSubject());
            log.info("postName: {}", emailMessage.getPostName());

            HashMap<String, String> params = new HashMap<>();
            params.put("postName", emailMessage.getPostName());
            params.put("postUrl", emailMessage.getPostUrl());
            params.put("commentContent", emailMessage.getCommentContent());
            params.put("commentAuthor", emailMessage.getCommentAuthor());

            emailService.sendEmail(MessageType.COMMENT, emailMessage.getEmail(), emailMessage.getSubject(), params);

        } catch (Exception e) {
            log.error("onMessage error : {}", e.getMessage());
        }
    }
}

File: backend/mail-service/src/main/kotlin/io/blog/mail/redis/service/CommentEmailSubService.kt
package io.blog.mail.redis.service

import com.fasterxml.jackson.databind.ObjectMapper
import io.blog.mail.mail.service.EmailService
import io.blog.mail.mail.types.MessageType
import io.blog.mail.redis.message.CommentEmailMessage
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.redis.connection.Message
import org.springframework.data.redis.connection.MessageListener
import org.springframework.stereotype.Service

@Service
class CommentEmailSubService : MessageListener {
    @Autowired
    private lateinit var emailService: EmailService
    private val mapper = ObjectMapper()

    override fun onMessage(message: Message, pattern: ByteArray?) {
        try {
            val emailMessage = mapper.readValue(message.body, CommentEmailMessage::class.java)

            val params = HashMap<String, String>()
            params["postName"] = emailMessage.postName
            params["postUrl"] = emailMessage.postUrl
            params["commentContent"] = emailMessage.commentContent
            params["commentAuthor"] = emailMessage.commentAuthor

            emailService.sendEmail(MessageType.COMMENT, emailMessage.email, emailMessage.subject, params)
        }
        catch (e: Exception) {
            e.printStackTrace()
        }
    }
}